src/NuGet.Services.AzureSearch/SearchService/IndexOperationBuilder.cs

```diff
@@ -63,14 +63,14 @@ public IndexOperation V2SearchWithHijackIndex(V2SearchRequest request)
             }
 
             var text = _textBuilder.Build(parsed);
-            var parameters = _parametersBuilder.V2Search(request);
+            var parameters = _parametersBuilder.V2Search(request, parsed.IsEmptyQuery());
```

Comment:
@loic-sharma I might have led @shishirx34 astray here -- do you think he should check for `*` for an empty grouping?

Reply:
looks like `SearchTextBuilder.Build` does return `*` when the `AzureSearchQueryBuilder` ends up being empty string, the check is after the empty grouping check. Not sure which scenarios this could be, but there is indeed a path that can be missed by simply checking the `parsed.IsEmptyQuery` vs `*` check for text.

https://github.com/NuGet/NuGet.Services.Metadata/blob/dev/src/NuGet.Services.AzureSearch/SearchService/SearchTextBuilder.cs#L107

https://github.com/NuGet/NuGet.Services.Metadata/blob/dev/src/NuGet.Services.AzureSearch/SearchService/SearchTextBuilder.cs#L150

Reply:
I think invalid scoped terms `foo:newtonsoft.json` would yield this. I guess we should do `== ThatConstantForAsterisk` instead of `IsEmptyQuery()`

Reply:
Sorry about the confusion -- maybe just dbl check with Loic to be sure. My brain is a little scattered right now.

Reply:
He should check for `*` due to the tags corner case: searching for `tags:,;|` will result in an non-empty grouping but an empty search query.